torch/testing/_internal/optests/generate_tests.py

```diff
@@ -493,10 +548,63 @@ def resolve_unique_overload_or_throw(op: torch._ops.OpOverloadPacket):
     return getattr(op, overload_name)
 
 
-def retrieve(failures_dict, qualname, test_name):
-    if qualname not in failures_dict:
-        return "success"
-    dct = failures_dict[qualname]
-    if test_name not in dct:
-        return "success"
-    return dct[test_name]
+DUMP_OPTIONS = {"indent": 2, "sort_keys": True}
+
+
+class FailuresDict:
+    def __init__(self, path, data):
+        self.path = path
+        self.data = data
+
+    @staticmethod
+    def load(path, *, create_file=False):
+        if create_file and not os.path.exists(path):
+            result = FailuresDict(path, {})
+            FailuresDict.save()
+            return result
+        with open(path) as fp:
+            dct = json.load(fp)
+        assert "data" in dct
+        assert "_version" in dct and dct["_version"] == 1
+        return FailuresDict(path, dct["data"])
+
+    def _save(self, to_str=False):
+        to_dump = {
+            "_description": (
+                f"This is a dict containing failures for tests autogenerated by "
+                f"generate_opcheck_tests. "
+                f"For more details, please see {GDOC}"
+            ),
+            "data": self.data,
+            "_version": 1,
+        }
+        if to_str:
+            return json.dumps(to_dump, **DUMP_OPTIONS)
+        with open(self.path, "w") as fp:
+            return json.dump(to_dump, fp, **DUMP_OPTIONS)
+
+    def save(self):
+        return self._save()
+
+    def get_status(self, qualname, test_name):
+        if qualname not in self.data:
+            return "success"
```

Comment:
interesting...

Reply:
I guess this feels less bad when you realize this is "expect success", as opposed to "yolo it succeeded"

Reply:
My mental model is that we're talking about the status of the test:
- "skip" means we should skip the test
- "xfail" means we expect the test to fail
- "success" means we expect the test to succeed.
I could rename this "xsuccess" if we want for consistency (do you have other thoughts?) Though I'll probably land this first to avoid merge conflicts